app/routes/task_routes.py

```diff
@@ -0,0 +1,109 @@
+from flask import Blueprint, request, make_response, jsonify, abort
+from app.models.task import Task
+from app import db
+from .routes_helpers import validate_model, delete_message
+from datetime import date
+import requests
+import os
+from dotenv import load_dotenv
+
+load_dotenv()
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+# CREATE TASK ENDPOINT
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+
+    if not request_body.get("title") or not request_body.get("description"):
+        abort(make_response({"details": "Invalid data"}, 400))
+
+    new_task = Task.from_dict(request_body)
+    
+    db.session.add(new_task)
+    db.session.commit()
+
+    return make_response({"task": new_task.to_dict()}, 201)
+
+# GET TASKS ENDPOINT
+@tasks_bp.route("", methods=["GET"])
+def read_tasks():
+    sort_query = request.args.get("sort")
+
+    if not sort_query:
+        tasks = Task.query.all()
+    elif sort_query.lower() == "asc":
+        tasks = Task.query.order_by(Task.title).all()
+    elif sort_query.lower() == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+
+    try:
+        tasks_response = [task.to_dict() for task in tasks]
+    except:
+        abort(make_response({"message": f"{sort_query} query invalid"}, 400))
```

Comment:
I see what you're doing here and it makes sense. What would an invalid query be? Like if someone passed `?sort=cats`? If so, then you're right that there will be an exception in trying to reference a variable `tasks` that isn't assigned, but I don't think you should let Python raise that exception. You could assign `tasks = []` before the if/else logic and then no tasks will be found and an empty list response would be returned. I think that aligns a bit better with expectations - if I request all the tasks and pass a bad sort parameter, I think expect no tasks to come back, rather than a 400.